### ISSUE TITLE
Cmake minimum version 3.8 is getting outdated.

### ISSUE THREAD

**Issue by christianbrugger** [asmjit/asmjit#450]
### Issue Description

When I configure `asmjit` with `cmake 3.31` I get the following warning:
```cmake
CMake Deprecation Warning at external/asmjit/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.
```

I suggest bumping the minimum required version to at least 3.10, currently its 3.8

### Operating System

Not specified / possibly all

### Architecture

Not specified

**Comment by kobalicek** [comment]
Yeah that makes sense.

I will check this out, I think the version can be raised even above that as I would not like to see a deprecation warning next year :)

**Comment by kobalicek** [comment]
Should be fixed by e7239626b8d5db1624faa08293fd470fe019cd88

Thanks!
